os/kernel/log_dump/log_dump.c

```diff
@@ -278,6 +283,14 @@ int log_dump_set(FAR const char *buffer, size_t buflen)
 		is_started_to_save = true;
 	} else if (strncmp(buffer, LOGDUMP_SAVE_STOP, strlen(LOGDUMP_SAVE_STOP) + 1) == 0) {
 		is_started_to_save = false;
+		/* Now we should compress the partially filled uncompressed buffer */
+		if (log_dump_read_wake() != OK) {
+			ldpdbg("ERROR: log dump read wake fail\n");
+			/* Should we return error from here?? because partially filled uncompressed buffer
+			 * might not be compressed but log_dump save has stopped now, so operation is success */
+		}
+	} else if (strncmp(buffer, LOGDUMP_GET_SIZE, strlen(LOGDUMP_GET_SIZE) + 1) == 0) {
```

Comment:
There are 3 strncmp s to check a command. It looks not good.
Because the data is a just character, how about using switch / case?